### DISCUSSION TITLE
get screen resolution in pixles

### DISCUSSION THREAD

**Discussion by NicoBiagi** [jspsych/jsPsych#2733] [Q&A]
Is there a way to get the resolution of the screen in pixels and add it to the  jsPsych.data.get()?

**Comment by nikbpetrov** [comment]
I think you can achieve that with the [browser-check](https://www.jspsych.org/7.3/plugins/browser-check/) plugin.

**Comment by NicoBiagi** [comment]
Thank you for your suggestion @nikbpetrov!

I have added the following to my scrip:

    <script src="https://unpkg.com/@jspsych/plugin-browser-check@1.0.1"></script>

   var screen_setting = {
        type: jsPsychBrowserCheck
    };

timeline.push(screen_settings);

But i get the following error message:

Uncaught ReferenceError: regeneratorRuntime is not defined
    at e.value (plugin-browser-check@1.0.1:1:7598)
    at e.value (plugin-browser-check@1.0.1:1:7516)
    at K.doTrial (jspsych@7.2.1:1:58625)
    at K.nextTrial (jspsych@7.2.1:1:57515)
    at K.finishTrial (jspsych@7.2.1:1:54109)
    at T (plugin-preload@1.1.0:1:3136)
    at _ (plugin-preload@1.1.0:1:2414)
    at A (plugin-preload@1.1.0:1:4400)
    at o.onload (jspsych@7.2.1:1:15665)

Do you know what I've done wrong?

Many thanks,
Nico

**Comment by nikbpetrov** [comment]
It's definitely related to how you've attempted to import the package. In all fairness, I am not familiar with this type of importing. I usually download the jsPsych library from the GitHub repos and import any relevant plugins via `<script src="jspsych/plugin-browser-check.js"></script>` (though make sure to point to the correct location of the script). To double-check that the importing is the source of your error, just download this one plugin and point to it via your local script.

Then the rest of what you have should work fine.

**Comment by bjoluc** [comment]
I guess the docs don't mention that jsPsych needs to be included before all plugins – maybe that's the problem?

**Reply by nikbpetrov** [reply]
Not importing the jspsych main script before plugins did not reproduce the error for me; instead it simply threw uncaught references in relation to module imports. I doubt this would be the cause as the error stack @NicoBiagi posted references the jspsych.js script so it is running, but it is getting stopped somewhere else. Unfortunately, it seems like he's using minified versions of the scripts so I can't easily locate where the error is coming from.

**Reply by bjoluc** [reply]
Thanks for checking @nikbpetrov! I should add that the reason for my comment was https://github.com/jspsych/jsPsych/pull/2643 and the regenerator runtime error only occurs with the CDN version...
